src/crimson/os/seastore/async_cleaner.cc

```diff
@@ -1659,6 +1662,7 @@ void RBMCleaner::commit_space_used(paddr_t addr, extent_len_t len)
   for (auto rbm : rbms) {
     if (addr.get_device_id() == rbm->get_device_id()) {
       if (rbm->get_start() <= addr) {
+	stats.used_bytes += len;
```

Comment:
Probably I'm missing something. Why should `commit_space_used()` increase the `used_bytes`? It doesn't match my understanding that in `AvlAllocator`, `complete_allocation()` only "confirms" the previous allocation, implying that the space is already allocated/used.

Reply:
In complete_commit(), commit_space_used() is called instead of mark_space_used() in order to commit fresh extents. Note that mark_space_used() is only used at mount() time in case of RBM.

Reply:
Ah, I see the relationships amoung `mark_extent_used()`, `alloc_extent()` and `complete_allocation()` in `ExtentAllocator`.

Reply:
How about moving this `stats.used_bytes += len` from `RBMCleaner::commit_space_used()` to `RBMCleaner::alloc_paddr()`, so that the special `RBMCleaner::mark_inflight_space_free()` is no longer necessary?

This way, the behavior of `stats.used_bytes` will be more consistent with the `ExtentAllocator`.

Reply:
If a conflict occurs, we should clear all allocated (not committed) extents. This means that we need to add stats.used_bytes -= len if we put stats.used_bytes in alloc_paddr().

Do you think this is more reasonable? I'm fine with this approach as well, but it looks like inconsistent information even though the extent is not committed.

Reply:
> If a conflict occurs, we should clear all allocated (not committed) extents. This means that we need to add stats.used_bytes -= len if we put stats.used_bytes in alloc_paddr().

It's already there, and with https://github.com/ceph/ceph/pull/50555#discussion_r1144467463, we don't need to change it to `mark_inflight_space_free()`, right?

Refer to https://github.com/ceph/ceph/pull/50555/commits/7f39c472019ef991fdf33cd5a6352724ba8f218a#diff-2d1edaa7a9eebb3a65135f5af648dd069934fa6bf6e6e0339b5771b0daef3727R874-R876

> but it looks like inconsistent information even though the extent is not committed

I think it's only about the decision whether we should include the "reserved allocations" in `stats.used_bytes` of `RBMCleaner`. I think both way should be fine for now until it is actually required to differentiate the reserved allocation usage. So just pick the simpler way with less modifications.

Reply:
OK.